traffic_portal/app/src/common/api/DeliveryServiceService.js

```diff
@@ -81,6 +81,41 @@ var DeliveryServiceService = function(Restangular, $http, $q, locationUtils, htt
         return deferred.promise;
     };
 
+    this.getServerCapabilities = function(id) {
+        return $http.get(ENV.api['root'] + 'deliveryservices_required_capabilities', { params: { deliveryServiceID: id } }).then(
+            function (result) {
+                return result.data.response;
+            },
+            function (err) {
+                throw err;
+            }
+        )
+    };
+
+    this.addServerCapability = function(deliveryServiceId, capabilityName) {
+        return $http.post(ENV.api['root'] + 'deliveryservices_required_capabilities', { deliveryServiceID: deliveryServiceId, requiredCapability: capabilityName}).then(
+            function(result) {
+                return result.data;
+            },
+            function(err) {
+                messageModel.setMessages(err.data.alerts, false);
```

Comment:
I know that we don't do this anywhere else, but since errors can occur at a lower level than the application layer, this `err` may not be an HTTP response at all. According to [the `$http` docs](https://docs.angularjs.org/api/ng/service/$http#$http-returns), you can check `err.status` is greater than zero to determine that. Or you could just check that `err.data` and `err.data.alerts` are both defined, since you probably want to check the latter in either case.